JniBridge.bee.cs

```diff
@@ -0,0 +1,379 @@
+        jdk.SetupJar(new NPath[] { classFileDir }, classFiles, apiGeneratorJar);
+        Backend.Current.AddAliasDependency(jarname, apiGeneratorJar);
+        return apiGeneratorJar;
+    }
+
+    struct SourceGenerationParams
+    {
+        public string platformName;
+        public NPath[] inputJars;
+        public string[] classes;
+    }
+
+    static SourceGenerationParams GetAndroidSourceGenerationParams(NPath sdk)
+    {
+        var androidJar = sdk.Combine("platforms", kAndroidApi, "android.jar");
+        var googlePlayServicesJar = sdk.Combine("extras", "google", "google_play_services_froyo", "libproject", "google-play-services_lib", "libs", "google-play-services.jar");
+
+        return new SourceGenerationParams()
+        {
+            platformName = "android",
+            inputJars = new[] {androidJar, googlePlayServicesJar},
+            classes = kAndroidApiClasses,
+        };
+    }
+
+    static SourceGenerationParams GetMacOSSourceGenerationParams(Jdk jdk)
+    {
+        var rtJar = jdk.JavaHome.Combine("jre", "lib", "rt.jar");
+
+        return new SourceGenerationParams()
+        {
+            platformName = "osx",
+            inputJars = new [] {rtJar},
+            classes = kMacOSClasses,
+        };
+    }
+    static NPath SetupSourceGeneration(Jdk jdk, NPath apiGenerator, NPath jnibridgeJar, SourceGenerationParams genParams)
+    {
+        var destDir = new NPath("artifacts").Combine("generated", genParams.platformName);
+        var actionName = "generate:jnibridge:" + genParams.platformName;
+        var generatedFiles = new NPath[] { destDir.Combine("API.h") };
+
+        var templateFiles = new NPath("templates").Files(new [] { "cpp", "h" });
+        var inputs = new List<NPath>();
+        inputs.AddRange(genParams.inputJars);
+        inputs.Add(jnibridgeJar);
+        inputs.Add(apiGenerator);
+        inputs.AddRange(templateFiles);
+
+        string inputJars = string.Empty;
+        if (genParams.inputJars.Length > 0)
+        {
+            var builder = new StringBuilder();
+            builder.Append('"').Append(genParams.inputJars[0]);
+            for (int i = 1; i < genParams.inputJars.Length; ++i)
+                builder.Append(';').Append(genParams.inputJars[i]);
+            inputJars = builder.Append('"').ToString();
+        }
+        var apiClassString = string.Join(" ", genParams.classes);
+        
+        Backend.Current.AddAction(
+            actionName,
+            generatedFiles,
+            inputs.ToArray(),
+            jdk.Java.InQuotes(SlashMode.Native),
+            new[]
+            {
+                "-XX:MaxPermSize=128M",
+                "-cp",
+                apiGenerator.InQuotes(),
+                "APIGenerator",
+                destDir.InQuotes(),
+                inputJars,
+                apiClassString
+            },
+            targetDirectories:new []{destDir}
+        );
+
+        Backend.Current.AddAliasDependency(actionName, generatedFiles);
```

Comment:
I don't know much about the build system, but is this needed, when we create the action with this?
Backend.Current.AddAction(
            actionName,
            generatedFiles,
Wouldn't the dependency be automatically created?